java/ql/src/semmle/code/java/Annotation.qll

```diff
@@ -60,16 +113,30 @@ class Annotation extends @annotation, Expr {
 
   /**
    * Gets a value of the annotation element with the specified `name`, which must be declared as an array
-   * type.
+   * type. This includes default values in case no explicit value is specified.
    *
    * If the annotation element is defined with an array initializer, then the returned value will
    * be one of the elements of that array. Otherwise, the returned value will be the single
    * expression defined for the value.
    */
-  Expr getAValue(string name) {
+  Expr getAValue(string name) { result = getAValue(name, _) }
+
+  /**
+   * Gets the value at a given index of the annotation element with the specified `name`, which must be
```

Comment:
The value or a value?

Reply:
Do you mean the mismatch between QLDoc ("the value") and predicate name ("getAValue")?
The predicate name should probably then be `getValue`, but the issue is that this makes the predicate names possibly even more confusing:
- `getAValue()`: Works only for annotations with single value, e.g. `@Target(TYPE)`
- `getValue(string)`: Works only for annotations with single value
- `getAValue(string)`: Works for annotations with single value and with array, e.g. `@Target(TYPE)`, `@Target({TYPE, METHOD)`

Naming the predicate `getValue(string, int)` might make you think at first that it is only for single values (which it is not). But maybe that is not such a big problem because the `int` parameter of the signature already gives a hint that it works for arrays, and you would have the QLDoc explaining its behavior anyways.

Should I change it to `getValue(string, int)` then?

Reply:
If the predicate is functional (for a given `name, index` pair there should be either one or zero results) then name it `getValue` and use `the` in the doc. Otherwise use `getAValue` and use `a` in the doc.

Reply:
Do you think it would make sense to deprecate `getAValue(string)` and instead rename it to `getAnArrayValue(string)`?
Then this newly added predicate could be named `getArrayValue(string, int)` which should hopefully make the distinction between the predicates more clear.

Reply:
Have renamed the predicates to `getAnArrayValue(string)` and `getArrayValue(string, int)` and have deprecated the old `getAValue(string)`.